crates/nu-command/src/network/url/parse.rs

```diff
@@ -0,0 +1,184 @@
+use super::url;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Value,
+};
+
+use url::Url;
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "url parse"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("url parse")
```

Comment:
You'll need to declare a vec of `(input_type, output_type)` pairs here -- lots of examples in the other command files. You'll see the text failure for your example until you declare these:

```
thread 'network::url::parse::test::test_examples' panicked at 'The example `echo 'http://www.example.com/foo/bar?p=section&s=&f[name]=vldc' | url parse` demonstrates a transformation of type String -> Record([("scheme", String), ("host", String), ("path", String), ("query", String), ("params", Record([("p", String), ("s", String), ("f[name]", String)]))]). However, this does not match the declared signature: []. (Did you forget to declare the input and output types for the command?) For this command, `vectorizes_over_list` is false and `operates_on_cell_paths()` is true.', crates/nu-command/src/example_test.rs:178:21
```